### PR TITLE
[CI] Added CI run on minimum and current python requirements

### PR DESCRIPTION
**Pull request by DjLegolas**
CI will run on both python 3.6 and 3.8

### CODE REVIEW

**Comment by DjLegolas** [comment]
Seem as there is no way to test on python3.6 because there is no libtorrent binding to 3.6 for windows or linux...

**Comment by doadin** [comment]
imo 3.10 has been working nicely, I think we could change this from 3.7-3.10.

**Comment by DjLegolas** [comment]
I'm thinking about upgrading the minimum python version, again, to 3.7 and run the CI on all the supported versions (3.7-3.10)...
@cas-- @gazpachoking @aresch what do you guys think?

**Comment by gazpachoking** [comment]
I'm 👍 on dropping 3.7 support. Not sure when we'd want to fit that in in terms of versioning and releases. In terms of this PR, I'd say testing more than one version is an improvement, regardless of whether it hits every currently supported version. I'm also 👍 on adding in other versions up to 3.10

**Comment by doadin** [comment]
should also maybe add fail-fast: false

**Comment by DjLegolas** [comment]
While testing, it helps having everything fail fast.
But will add that flag later, thanks @doadin

**Comment by doadin** [comment]
3.10 needs pytest >= 6.2.5 I think.

**Comment by doadin** [comment]
https://github.com/deluge-torrent/deluge/blob/develop/requirements-tests.txt#L1
pytest == 6.2.5 ; python_version >= '3.10'
pytest != 5.2.3, < 5.4
maybe

**Comment by DjLegolas** [comment]
No, I'll just drop the limits from `pytest`...
It worked on my local Windows and Ubuntu, both with python 3.8.

**Review by doadin — APPROVED** [review]

**Comment by DjLegolas** [comment]
To sums this up:
* Testing python versions 3.7-3.10
* Switched `libtorrent`'s installation source to `pypi`
* Dropped `pytest` version limits, current version is `7.0.0` for python 3.10 (thanks @doadin for the heads up)

**Comment by doadin** [comment]
I am noticing a few things we use like twisted in a coming version is droping python3.6 and pyinstaller, Pillow already has, could probabbly drop 3.6 our selves as well. if it was only pyinstaller then id say no but deps are starting to drop as well. At least on the windows side requiring 3.7+ only excluded windows xp I think which nobody should still be using so not really a big deal imo.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@ on:
 jobs:
   test-linux:
     runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
```
Comment (cas--) [inline .github/workflows/ci.yml:14]: Let's keep fail-fast on

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,6 +10,10 @@ on:
...
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```
Comment (cas--) [inline .github/workflows/ci.yml:16]: can we test just 3.7 and 3.10

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -32,22 +36,14 @@ jobs:
...
-          install: python3-libtorrent-dbg
-
       - name: Sets env var for security
         if: (github.event_name == 'pull_request' && contains(github.event.pull_request.body, 'security_test')) || (github.event_name == 'push' && contains(github.event.head_commit.message, 'security_test'))
         run: echo "SECURITY_TESTS=True" >> $GITHUB_ENV
 
       - name: Install dependencies
         run: |
           pip install --upgrade pip wheel
+          pip install libtorrent==1.2.*
```
Comment (cas--) [inline .github/workflows/ci.yml:46]: let's move this to the requirements.txt and remove the version requirement

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -68,7 +64,7 @@ jobs:
       - name: Test with pytest
         run: |
           ulimit -c unlimited  # Enable core dumps to be captured
-          cp /usr/lib/python3/dist-packages/libtorrent*.so $GITHUB_WORKSPACE/deluge
+          cp $LD_LIBRARY_PATH/python3*/site-packages/libtorrent*.so $GITHUB_WORKSPACE/deluge
```
Comment (cas--) [inline .github/workflows/ci.yml:67]: this lib copy should not be needed if using pip install

**Review by cas-- — COMMENTED** [review]
A few suggestions to make to both linux and win runs

File: .github/workflows/ci.yml (hunk 3 above)
Comment (DjLegolas) [reply .github/workflows/ci.yml:46]: Just a reminder, removing the version requirement will install libtorrent 2

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DjLegolas) [reply .github/workflows/ci.yml:14]: Done

File: .github/workflows/ci.yml (hunk 4 above)
Comment (DjLegolas) [reply .github/workflows/ci.yml]: Removed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cas--) [reply .github/workflows/ci.yml:14]: What I meant was, can we remove this line

**Pull request closed without merging**